File: 0078-subsets/0078-subsets.java
class Solution {
    public void Dfs(List<List<Integer>> res, List<Integer> cur, int[] nums, int i){
        if (i == nums.length){
            res.add(new ArrayList<Integer>(cur));
            return;
        }

        cur.add(nums[i]);
        Dfs(res, cur, nums, i+1);

        cur.remove(cur.size() - 1);
        Dfs(res, cur, nums, i+1);

    }

    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> cur = new ArrayList<>();
        Dfs(res, cur, nums, 0);
        return res;
    }
}

File: 0078-subsets/0078-subsets.cs
public class Solution {
    public void Dfs(int i, IList<IList<int>> res, int[] nums, IList<int> cur){
        if (i == nums.Length){
            res.Add(new List<int>(cur));
            return;
        }
        cur.Add(nums[i]);
        Dfs(i+1, res, nums, cur);

        cur.RemoveAt(cur.Count - 1);
        Dfs(i+1, res, nums, cur);

    }   
    public IList<IList<int>> Subsets(int[] nums) {
        var res = new List<IList<int>> ();
        var cur = new List<int>();
        Dfs(0, res, nums, cur);

        return new List<IList<int>> (res);
    }
}